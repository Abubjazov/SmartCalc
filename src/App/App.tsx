export const App = () => {
	return (
		<div className='App'>
			<h1>Hello!</h1>
		</div>
	)
}
