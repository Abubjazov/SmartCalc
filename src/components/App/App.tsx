import { lazy, Suspense } from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'

import { MainPage } from '../../pages'
import { Header } from '../Header/Header'
import { Spinner } from '../Spinner/Spinner'

const LoginPage = lazy(() =>
	import('../../pages').then(({ LoginPage }) => ({ default: LoginPage }))
)

const CalcPage = lazy(() =>
	import('../../pages').then(({ CalcPage }) => ({ default: CalcPage }))
)

const Page404 = lazy(() =>
	import('../../pages').then(({ Page404 }) => ({ default: Page404 }))
)

export const App = (): JSX.Element => {
	return (
		<BrowserRouter>
			<Header />
			<Suspense fallback={<Spinner />}>
				<Routes>
					<Route path='/smartcalc/' element={<MainPage />} />
					<Route path='/smartcalc/login' element={<LoginPage />} />
					<Route path='/smartcalc/calc' element={<CalcPage />} />
					<Route path='*' element={<Page404 />} />
				</Routes>
			</Suspense>
		</BrowserRouter>
	)
}
