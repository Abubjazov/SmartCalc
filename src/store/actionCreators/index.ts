/* eslint-disable import/no-anonymous-default-export */
import * as CalcActionCreators from './calc.actionCreators'
import * as AuthActionCreators from './auth.actionCreators'

export default {
	...CalcActionCreators,
	...AuthActionCreators,
}
