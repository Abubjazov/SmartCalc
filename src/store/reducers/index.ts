import { combineReducers } from 'redux'
import { authReducer } from './authReducer'
import { calcReducer } from './calcReducer'

export const rootReducer = combineReducers({
	calc: calcReducer,
	auth: authReducer,
})

export type RootState = ReturnType<typeof rootReducer>
