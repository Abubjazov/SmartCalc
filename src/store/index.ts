import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
// import { composeWithDevTools } from '@redux-devtools/extension'

import { rootReducer } from './reducers'

export const store = createStore(rootReducer, applyMiddleware(thunk))

// export const store = createStore(
// 	rootReducer,
// 	composeWithDevTools(applyMiddleware(thunk))
// )
