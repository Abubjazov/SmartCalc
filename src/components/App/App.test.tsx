import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import { App } from './App'
import { store } from '../../store'

describe('Component: App', () => {
	test('renders App', () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<App />
				</Provider>
			)
			.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
