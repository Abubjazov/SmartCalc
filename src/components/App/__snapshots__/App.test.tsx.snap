// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App renders App 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <svg
        className="qiwi-logo"
        focusable="false"
        viewBox="0 0 80 40"
      >
        <path
          d="M15.9536232,-3.55271368e-15 C24.7652174,-3.55271368e-15 31.9072464,7.14202899 31.9072464,15.9536232 C31.9072464,18.9449275 31.0724638,21.7507246 29.657971,24.1391304 C29.6115942,24.2086957 29.4956522,24.1855072 29.4724638,24.0927536 C28.915942,20.173913 26.5275362,18.0173913 23.0492754,17.3681159 C22.7478261,17.3217391 22.7014493,17.1362319 23.0956522,17.0898551 C24.1623188,16.9971014 25.6695652,17.0202899 26.457971,17.1594203 C26.5043478,16.7652174 26.5275362,16.3478261 26.5275362,15.9304348 C26.5275362,10.1101449 21.7971014,5.37971014 15.9768116,5.37971014 C10.1565217,5.37971014 5.42608696,10.1101449 5.42608696,15.9304348 C5.42608696,21.7507246 10.1565217,26.4811594 15.9768116,26.4811594 L16.4637681,26.4811594 C16.2782609,25.4608696 16.2318841,24.4173913 16.2550725,23.6289855 C16.2782609,22.9797101 16.4173913,22.8869565 16.6956522,23.3971014 C18.1565217,25.9246377 20.2434783,28.1971014 24.3246377,29.1014493 C27.6637681,29.8434783 31.0028986,30.7014493 34.5971014,35.2695652 C34.9217391,35.6637681 34.4347826,36.0811594 34.0637681,35.7565217 C30.4,32.5101449 27.0608696,31.4434783 24.0231884,31.4434783 C20.6144928,31.4666667 18.2956522,31.9072464 15.9536232,31.9072464 C7.14202899,31.9072464 0,24.7652174 0,15.9536232 C0,7.14202899 7.14202899,-3.55271368e-15 15.9536232,-3.55271368e-15 Z M24.0695652,25.2057971 C24.4405797,25.2057971 24.857971,25.3681159 25.2289855,25.6927536 C25.9478261,26.2956522 26.1565217,26.9913043 25.7855072,27.5014493 C25.5768116,27.7565217 25.2289855,27.9188406 24.8347826,27.9188406 C24.4405797,27.9188406 24.0231884,27.7797101 23.7449275,27.5246377 C23.0956522,26.9681159 22.9101449,26.0405797 23.3275362,25.5304348 C23.4898551,25.3217391 23.7449275,25.2057971 24.0695652,25.2057971 Z M27.5942029,22.5855072 C28.1971014,22.8173913 28.4289855,23.7217391 28.4753623,24.115942 C28.5681159,24.8347826 28.3594203,25.1130435 28.1275362,25.1130435 C27.8956522,25.1130435 27.5710145,24.8347826 27.2231884,24.2782609 C26.8753623,23.7217391 26.7362319,23.0956522 26.9217391,22.7710145 C27.0376812,22.5623188 27.2927536,22.4695652 27.5942029,22.5855072 Z M44.6144928,4.66086957 C48.1855072,4.66086957 51.084058,7.55942029 51.084058,11.1304348 C51.084058,12.4057971 50.7130435,13.6115942 50.0637681,14.6318841 L52.0115942,16.9043478 C52.1507246,17.0666667 52.0347826,17.2985507 51.826087,17.2985507 L48.9043478,17.2985507 C48.7884058,17.2985507 48.6724638,17.2521739 48.6028986,17.1594203 L48.115942,16.5565217 C47.0956522,17.2057971 45.8898551,17.6 44.6144928,17.6 C41.0434783,17.6 38.1449275,14.7014493 38.1449275,11.1304348 C38.1449275,7.55942029 41.0434783,4.66086957 44.6144928,4.66086957 Z M56.115942,4.96231884 C56.3014493,4.96231884 56.4405797,5.10144928 56.4405797,5.28695652 L56.4405797,16.9971014 C56.4405797,17.1826087 56.3014493,17.3217391 56.115942,17.3217391 L53.7275362,17.3217391 C53.542029,17.3217391 53.4028986,17.1826087 53.4028986,16.9971014 L53.4028986,5.28695652 C53.4028986,5.10144928 53.542029,4.96231884 53.7275362,4.96231884 L56.115942,4.96231884 Z M60.4753623,4.96231884 C60.5913043,4.96231884 60.6911433,5.02673108 60.7480408,5.12871712 L60.7768116,5.1942029 L62.8869565,12.1275362 L65.2057971,5.1942029 C65.2428986,5.10144928 65.3245217,5.02353623 65.4269217,4.99608116 L65.5072464,4.98550725 L67.3855072,4.98550725 C67.4968116,4.98550725 67.5932754,5.03002899 67.6511536,5.11907246 L67.6869565,5.1942029 L70.0057971,12.1275362 L72.115942,5.1942029 C72.1545894,5.07826087 72.2415459,4.99452496 72.3499732,4.9698336 L72.4173913,4.96231884 L75.084058,4.96231884 C75.1768116,4.96231884 75.2927536,5.00869565 75.4086957,5.10144928 C75.4643478,5.15710145 75.4754783,5.24243478 75.4658319,5.32183188 L75.4550725,5.37971014 L71.5130435,17.1130435 C71.475942,17.2057971 71.3943188,17.2837101 71.2919188,17.3111652 L71.2115942,17.3217391 L69.0782609,17.3217391 C68.9669565,17.3217391 68.8704928,17.2772174 68.8126145,17.1881739 L68.7768116,17.1130435 L66.4811594,10.2724638 L64.1855072,17.1130435 C64.1484058,17.2057971 64.0667826,17.2837101 63.9643826,17.3111652 L63.884058,17.3217391 L61.7507246,17.3217391 C61.6394203,17.3217391 61.5429565,17.2772174 61.4850783,17.1881739 L61.4492754,17.1130435 L57.5072464,5.37971014 C57.484058,5.28695652 57.484058,5.1942029 57.5536232,5.10144928 C57.5907246,5.02724638 57.6723478,4.98272464 57.7510029,4.96788406 L57.8086957,4.96231884 L60.4753623,4.96231884 Z M79.1188406,4.96231884 C79.257971,4.96231884 79.3800651,5.06453712 79.4267123,5.21056323 L79.4434783,5.28695652 L79.4434783,16.9971014 C79.4434783,17.1594203 79.3369565,17.2862319 79.1860507,17.3153986 L79.1188406,17.3217391 L76.7304348,17.3217391 C76.5681159,17.3217391 76.4413043,17.2152174 76.4121377,17.0643116 L76.4057971,16.9971014 L76.4057971,5.28695652 C76.4057971,5.12463768 76.5123188,4.99782609 76.6632246,4.96865942 L76.7304348,4.96231884 L79.1188406,4.96231884 Z M44.6144928,7.53623188 C42.6434783,7.53623188 41.2057971,9.2057971 41.2057971,11.1304348 C41.2057971,13.0550725 42.6434783,14.7014493 44.6144928,14.7014493 C45.1710145,14.7246377 45.7275362,14.5623188 46.2144928,14.3072464 L44.9391304,12.7536232 C44.8,12.5913043 44.915942,12.3362319 45.1246377,12.3362319 L47.8144928,12.3362319 C47.9536232,11.9652174 48.0231884,11.5478261 48.0231884,11.1304348 C48.0231884,9.2057971 46.5855072,7.53623188 44.6144928,7.53623188 Z"
          fill="#FF8C00"
        />
      </svg>
      <p
        className="logo-text"
      >
        SMART CALC
      </p>
    </div>
    <nav
      className="header-menu"
    >
      <div
        className="hamburger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
      <ul>
        <li>
          <a
            aria-label="На главную"
            className=""
            href="/smartcalc/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="in-out"
    >
      <button
        onClick={[Function]}
      >
        Войти
      </button>
    </div>
  </header>,
  <svg
    className="spinner"
    height="200"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
    width="200"
  >
    <path
      d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
      fill="none"
      stroke="#cccccc"
      strokeDasharray="42.76482137044271 42.76482137044271"
      strokeLinecap="round"
      strokeWidth="8"
    >
      <animate
        attributeName="stroke-dashoffset"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="0;256.58892822265625"
      />
    </path>
  </svg>,
]
`;

exports[`Component: App renders App then token is NOT NULL 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <svg
        className="qiwi-logo"
        focusable="false"
        viewBox="0 0 80 40"
      >
        <path
          d="M15.9536232,-3.55271368e-15 C24.7652174,-3.55271368e-15 31.9072464,7.14202899 31.9072464,15.9536232 C31.9072464,18.9449275 31.0724638,21.7507246 29.657971,24.1391304 C29.6115942,24.2086957 29.4956522,24.1855072 29.4724638,24.0927536 C28.915942,20.173913 26.5275362,18.0173913 23.0492754,17.3681159 C22.7478261,17.3217391 22.7014493,17.1362319 23.0956522,17.0898551 C24.1623188,16.9971014 25.6695652,17.0202899 26.457971,17.1594203 C26.5043478,16.7652174 26.5275362,16.3478261 26.5275362,15.9304348 C26.5275362,10.1101449 21.7971014,5.37971014 15.9768116,5.37971014 C10.1565217,5.37971014 5.42608696,10.1101449 5.42608696,15.9304348 C5.42608696,21.7507246 10.1565217,26.4811594 15.9768116,26.4811594 L16.4637681,26.4811594 C16.2782609,25.4608696 16.2318841,24.4173913 16.2550725,23.6289855 C16.2782609,22.9797101 16.4173913,22.8869565 16.6956522,23.3971014 C18.1565217,25.9246377 20.2434783,28.1971014 24.3246377,29.1014493 C27.6637681,29.8434783 31.0028986,30.7014493 34.5971014,35.2695652 C34.9217391,35.6637681 34.4347826,36.0811594 34.0637681,35.7565217 C30.4,32.5101449 27.0608696,31.4434783 24.0231884,31.4434783 C20.6144928,31.4666667 18.2956522,31.9072464 15.9536232,31.9072464 C7.14202899,31.9072464 0,24.7652174 0,15.9536232 C0,7.14202899 7.14202899,-3.55271368e-15 15.9536232,-3.55271368e-15 Z M24.0695652,25.2057971 C24.4405797,25.2057971 24.857971,25.3681159 25.2289855,25.6927536 C25.9478261,26.2956522 26.1565217,26.9913043 25.7855072,27.5014493 C25.5768116,27.7565217 25.2289855,27.9188406 24.8347826,27.9188406 C24.4405797,27.9188406 24.0231884,27.7797101 23.7449275,27.5246377 C23.0956522,26.9681159 22.9101449,26.0405797 23.3275362,25.5304348 C23.4898551,25.3217391 23.7449275,25.2057971 24.0695652,25.2057971 Z M27.5942029,22.5855072 C28.1971014,22.8173913 28.4289855,23.7217391 28.4753623,24.115942 C28.5681159,24.8347826 28.3594203,25.1130435 28.1275362,25.1130435 C27.8956522,25.1130435 27.5710145,24.8347826 27.2231884,24.2782609 C26.8753623,23.7217391 26.7362319,23.0956522 26.9217391,22.7710145 C27.0376812,22.5623188 27.2927536,22.4695652 27.5942029,22.5855072 Z M44.6144928,4.66086957 C48.1855072,4.66086957 51.084058,7.55942029 51.084058,11.1304348 C51.084058,12.4057971 50.7130435,13.6115942 50.0637681,14.6318841 L52.0115942,16.9043478 C52.1507246,17.0666667 52.0347826,17.2985507 51.826087,17.2985507 L48.9043478,17.2985507 C48.7884058,17.2985507 48.6724638,17.2521739 48.6028986,17.1594203 L48.115942,16.5565217 C47.0956522,17.2057971 45.8898551,17.6 44.6144928,17.6 C41.0434783,17.6 38.1449275,14.7014493 38.1449275,11.1304348 C38.1449275,7.55942029 41.0434783,4.66086957 44.6144928,4.66086957 Z M56.115942,4.96231884 C56.3014493,4.96231884 56.4405797,5.10144928 56.4405797,5.28695652 L56.4405797,16.9971014 C56.4405797,17.1826087 56.3014493,17.3217391 56.115942,17.3217391 L53.7275362,17.3217391 C53.542029,17.3217391 53.4028986,17.1826087 53.4028986,16.9971014 L53.4028986,5.28695652 C53.4028986,5.10144928 53.542029,4.96231884 53.7275362,4.96231884 L56.115942,4.96231884 Z M60.4753623,4.96231884 C60.5913043,4.96231884 60.6911433,5.02673108 60.7480408,5.12871712 L60.7768116,5.1942029 L62.8869565,12.1275362 L65.2057971,5.1942029 C65.2428986,5.10144928 65.3245217,5.02353623 65.4269217,4.99608116 L65.5072464,4.98550725 L67.3855072,4.98550725 C67.4968116,4.98550725 67.5932754,5.03002899 67.6511536,5.11907246 L67.6869565,5.1942029 L70.0057971,12.1275362 L72.115942,5.1942029 C72.1545894,5.07826087 72.2415459,4.99452496 72.3499732,4.9698336 L72.4173913,4.96231884 L75.084058,4.96231884 C75.1768116,4.96231884 75.2927536,5.00869565 75.4086957,5.10144928 C75.4643478,5.15710145 75.4754783,5.24243478 75.4658319,5.32183188 L75.4550725,5.37971014 L71.5130435,17.1130435 C71.475942,17.2057971 71.3943188,17.2837101 71.2919188,17.3111652 L71.2115942,17.3217391 L69.0782609,17.3217391 C68.9669565,17.3217391 68.8704928,17.2772174 68.8126145,17.1881739 L68.7768116,17.1130435 L66.4811594,10.2724638 L64.1855072,17.1130435 C64.1484058,17.2057971 64.0667826,17.2837101 63.9643826,17.3111652 L63.884058,17.3217391 L61.7507246,17.3217391 C61.6394203,17.3217391 61.5429565,17.2772174 61.4850783,17.1881739 L61.4492754,17.1130435 L57.5072464,5.37971014 C57.484058,5.28695652 57.484058,5.1942029 57.5536232,5.10144928 C57.5907246,5.02724638 57.6723478,4.98272464 57.7510029,4.96788406 L57.8086957,4.96231884 L60.4753623,4.96231884 Z M79.1188406,4.96231884 C79.257971,4.96231884 79.3800651,5.06453712 79.4267123,5.21056323 L79.4434783,5.28695652 L79.4434783,16.9971014 C79.4434783,17.1594203 79.3369565,17.2862319 79.1860507,17.3153986 L79.1188406,17.3217391 L76.7304348,17.3217391 C76.5681159,17.3217391 76.4413043,17.2152174 76.4121377,17.0643116 L76.4057971,16.9971014 L76.4057971,5.28695652 C76.4057971,5.12463768 76.5123188,4.99782609 76.6632246,4.96865942 L76.7304348,4.96231884 L79.1188406,4.96231884 Z M44.6144928,7.53623188 C42.6434783,7.53623188 41.2057971,9.2057971 41.2057971,11.1304348 C41.2057971,13.0550725 42.6434783,14.7014493 44.6144928,14.7014493 C45.1710145,14.7246377 45.7275362,14.5623188 46.2144928,14.3072464 L44.9391304,12.7536232 C44.8,12.5913043 44.915942,12.3362319 45.1246377,12.3362319 L47.8144928,12.3362319 C47.9536232,11.9652174 48.0231884,11.5478261 48.0231884,11.1304348 C48.0231884,9.2057971 46.5855072,7.53623188 44.6144928,7.53623188 Z"
          fill="#FF8C00"
        />
      </svg>
      <p
        className="logo-text"
      >
        SMART CALC
      </p>
    </div>
    <nav
      className="header-menu"
    >
      <div
        className="hamburger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
      <ul>
        <li>
          <a
            aria-label="На главную"
            className=""
            href="/smartcalc/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
        <li>
          <a
            aria-label="Калькулятор"
            className=""
            href="/smartcalc/calc"
            onClick={[Function]}
          >
            Калькулятор
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="in-out"
    >
      <button
        onClick={[Function]}
      >
        Выйти
      </button>
    </div>
  </header>,
  <main
    className="page-404"
  >
    <img
      alt="An error has occurred"
      src="Error.gif"
      style={
        Object {
          "display": "block",
          "height": "102px",
          "margin": "0 auto",
          "objectFit": "contain",
          "width": "128px",
        }
      }
    />
    <p
      role="alert"
      tabIndex={0}
    >
      Страница не найдена
    </p>
    <a
      className=""
      href="/smartcalc/"
      onClick={[Function]}
      role="alert"
    >
      Вернуться на главную
    </a>
  </main>,
]
`;

exports[`Component: App renders App then token is NULL 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logo"
    >
      <svg
        className="qiwi-logo"
        focusable="false"
        viewBox="0 0 80 40"
      >
        <path
          d="M15.9536232,-3.55271368e-15 C24.7652174,-3.55271368e-15 31.9072464,7.14202899 31.9072464,15.9536232 C31.9072464,18.9449275 31.0724638,21.7507246 29.657971,24.1391304 C29.6115942,24.2086957 29.4956522,24.1855072 29.4724638,24.0927536 C28.915942,20.173913 26.5275362,18.0173913 23.0492754,17.3681159 C22.7478261,17.3217391 22.7014493,17.1362319 23.0956522,17.0898551 C24.1623188,16.9971014 25.6695652,17.0202899 26.457971,17.1594203 C26.5043478,16.7652174 26.5275362,16.3478261 26.5275362,15.9304348 C26.5275362,10.1101449 21.7971014,5.37971014 15.9768116,5.37971014 C10.1565217,5.37971014 5.42608696,10.1101449 5.42608696,15.9304348 C5.42608696,21.7507246 10.1565217,26.4811594 15.9768116,26.4811594 L16.4637681,26.4811594 C16.2782609,25.4608696 16.2318841,24.4173913 16.2550725,23.6289855 C16.2782609,22.9797101 16.4173913,22.8869565 16.6956522,23.3971014 C18.1565217,25.9246377 20.2434783,28.1971014 24.3246377,29.1014493 C27.6637681,29.8434783 31.0028986,30.7014493 34.5971014,35.2695652 C34.9217391,35.6637681 34.4347826,36.0811594 34.0637681,35.7565217 C30.4,32.5101449 27.0608696,31.4434783 24.0231884,31.4434783 C20.6144928,31.4666667 18.2956522,31.9072464 15.9536232,31.9072464 C7.14202899,31.9072464 0,24.7652174 0,15.9536232 C0,7.14202899 7.14202899,-3.55271368e-15 15.9536232,-3.55271368e-15 Z M24.0695652,25.2057971 C24.4405797,25.2057971 24.857971,25.3681159 25.2289855,25.6927536 C25.9478261,26.2956522 26.1565217,26.9913043 25.7855072,27.5014493 C25.5768116,27.7565217 25.2289855,27.9188406 24.8347826,27.9188406 C24.4405797,27.9188406 24.0231884,27.7797101 23.7449275,27.5246377 C23.0956522,26.9681159 22.9101449,26.0405797 23.3275362,25.5304348 C23.4898551,25.3217391 23.7449275,25.2057971 24.0695652,25.2057971 Z M27.5942029,22.5855072 C28.1971014,22.8173913 28.4289855,23.7217391 28.4753623,24.115942 C28.5681159,24.8347826 28.3594203,25.1130435 28.1275362,25.1130435 C27.8956522,25.1130435 27.5710145,24.8347826 27.2231884,24.2782609 C26.8753623,23.7217391 26.7362319,23.0956522 26.9217391,22.7710145 C27.0376812,22.5623188 27.2927536,22.4695652 27.5942029,22.5855072 Z M44.6144928,4.66086957 C48.1855072,4.66086957 51.084058,7.55942029 51.084058,11.1304348 C51.084058,12.4057971 50.7130435,13.6115942 50.0637681,14.6318841 L52.0115942,16.9043478 C52.1507246,17.0666667 52.0347826,17.2985507 51.826087,17.2985507 L48.9043478,17.2985507 C48.7884058,17.2985507 48.6724638,17.2521739 48.6028986,17.1594203 L48.115942,16.5565217 C47.0956522,17.2057971 45.8898551,17.6 44.6144928,17.6 C41.0434783,17.6 38.1449275,14.7014493 38.1449275,11.1304348 C38.1449275,7.55942029 41.0434783,4.66086957 44.6144928,4.66086957 Z M56.115942,4.96231884 C56.3014493,4.96231884 56.4405797,5.10144928 56.4405797,5.28695652 L56.4405797,16.9971014 C56.4405797,17.1826087 56.3014493,17.3217391 56.115942,17.3217391 L53.7275362,17.3217391 C53.542029,17.3217391 53.4028986,17.1826087 53.4028986,16.9971014 L53.4028986,5.28695652 C53.4028986,5.10144928 53.542029,4.96231884 53.7275362,4.96231884 L56.115942,4.96231884 Z M60.4753623,4.96231884 C60.5913043,4.96231884 60.6911433,5.02673108 60.7480408,5.12871712 L60.7768116,5.1942029 L62.8869565,12.1275362 L65.2057971,5.1942029 C65.2428986,5.10144928 65.3245217,5.02353623 65.4269217,4.99608116 L65.5072464,4.98550725 L67.3855072,4.98550725 C67.4968116,4.98550725 67.5932754,5.03002899 67.6511536,5.11907246 L67.6869565,5.1942029 L70.0057971,12.1275362 L72.115942,5.1942029 C72.1545894,5.07826087 72.2415459,4.99452496 72.3499732,4.9698336 L72.4173913,4.96231884 L75.084058,4.96231884 C75.1768116,4.96231884 75.2927536,5.00869565 75.4086957,5.10144928 C75.4643478,5.15710145 75.4754783,5.24243478 75.4658319,5.32183188 L75.4550725,5.37971014 L71.5130435,17.1130435 C71.475942,17.2057971 71.3943188,17.2837101 71.2919188,17.3111652 L71.2115942,17.3217391 L69.0782609,17.3217391 C68.9669565,17.3217391 68.8704928,17.2772174 68.8126145,17.1881739 L68.7768116,17.1130435 L66.4811594,10.2724638 L64.1855072,17.1130435 C64.1484058,17.2057971 64.0667826,17.2837101 63.9643826,17.3111652 L63.884058,17.3217391 L61.7507246,17.3217391 C61.6394203,17.3217391 61.5429565,17.2772174 61.4850783,17.1881739 L61.4492754,17.1130435 L57.5072464,5.37971014 C57.484058,5.28695652 57.484058,5.1942029 57.5536232,5.10144928 C57.5907246,5.02724638 57.6723478,4.98272464 57.7510029,4.96788406 L57.8086957,4.96231884 L60.4753623,4.96231884 Z M79.1188406,4.96231884 C79.257971,4.96231884 79.3800651,5.06453712 79.4267123,5.21056323 L79.4434783,5.28695652 L79.4434783,16.9971014 C79.4434783,17.1594203 79.3369565,17.2862319 79.1860507,17.3153986 L79.1188406,17.3217391 L76.7304348,17.3217391 C76.5681159,17.3217391 76.4413043,17.2152174 76.4121377,17.0643116 L76.4057971,16.9971014 L76.4057971,5.28695652 C76.4057971,5.12463768 76.5123188,4.99782609 76.6632246,4.96865942 L76.7304348,4.96231884 L79.1188406,4.96231884 Z M44.6144928,7.53623188 C42.6434783,7.53623188 41.2057971,9.2057971 41.2057971,11.1304348 C41.2057971,13.0550725 42.6434783,14.7014493 44.6144928,14.7014493 C45.1710145,14.7246377 45.7275362,14.5623188 46.2144928,14.3072464 L44.9391304,12.7536232 C44.8,12.5913043 44.915942,12.3362319 45.1246377,12.3362319 L47.8144928,12.3362319 C47.9536232,11.9652174 48.0231884,11.5478261 48.0231884,11.1304348 C48.0231884,9.2057971 46.5855072,7.53623188 44.6144928,7.53623188 Z"
          fill="#FF8C00"
        />
      </svg>
      <p
        className="logo-text"
      >
        SMART CALC
      </p>
    </div>
    <nav
      className="header-menu"
    >
      <div
        className="hamburger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
      <ul>
        <li>
          <a
            aria-label="На главную"
            className=""
            href="/smartcalc/"
            onClick={[Function]}
          >
            Главная
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="in-out"
    >
      <button
        onClick={[Function]}
      >
        Войти
      </button>
    </div>
  </header>,
  <svg
    className="spinner"
    height="200"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
    width="200"
  >
    <path
      d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
      fill="none"
      stroke="#cccccc"
      strokeDasharray="42.76482137044271 42.76482137044271"
      strokeLinecap="round"
      strokeWidth="8"
    >
      <animate
        attributeName="stroke-dashoffset"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="0;256.58892822265625"
      />
    </path>
  </svg>,
]
`;
